src/app/products/products.component.ts

```diff
@@ -1,23 +1,41 @@
+import { switchMap } from 'rxjs/operators';
+import { Category } from './../models/category';
 import { Observable } from 'rxjs';
 import { CategoryService } from './../category.service';
 import { ProductService } from './../product.service';
 import { Component } from '@angular/core';
 import { Product } from '../models/product';
-import { Category } from '../models/category';
+import { ActivatedRoute } from '@angular/router';
 
 @Component({
   selector: 'app-products',
   templateUrl: './products.component.html',
 })
 export class ProductsComponent {
-  productsObservable: Observable<Product[]>;
+  products: Product[] = [];
+  filteredProducts: Product[];
   categoriesObservable: Observable<Category[]>;
+  paramCategory: string;
 
   constructor(
+    route: ActivatedRoute,
     productService: ProductService,
     categoryService: CategoryService,
   ) {
-    this.productsObservable = productService.getProductsObservable();
+    productService
+      .getProductsObservable()
+      .pipe(
+        switchMap((products) => {
+          this.products = products;
+          return route.queryParamMap;
+        }),
+      )
+      .subscribe((params) => {
+        this.paramCategory = params.get('category');
+        this.filteredProducts = this.paramCategory
+          ? this.products.filter((p) => p.category === this.paramCategory)
```

Comment:
no abbreviations

Reply:
the reason i say that is because it makes code harder to read but ultimately it's your call